### ISSUE TITLE
I don't like the UserConfigBuilder

### ISSUE THREAD

**Issue by museun** [museun/twitchchat#34]
It should be less verbose or maybe provide some 'canned' configs where the user just has to provide their nick and their oauth token.

```rust
    let config = twitchchat::UserConfig::builder()
        .commands()
        .membership()
        .tags()
        .nick(&opts.nick)
        .token(pass)
        .build()
        .expect("valid config"); // why is this an option?
```
could just be
```rust
    let config = UserConfig::all_caps()
        .with_nick("foo")
        .with_token("bar")
        .build();
```

And instead of returning an Option (which I assume most people would just unwrap anyway), panic because this should be a hard error. The UserConfig type won't be built dynamically, so if a required field is missing (nick and/or token) it should yell at the user rather then continuing.

**Comment by twitlydoof** [comment]
I agree.

**Comment by museun** [comment]
closed with b2f52b0
